app-editors/vscode/vscode-1.56.0.ebuild

```diff
@@ -15,7 +15,158 @@ SRC_URI="
 
 RESTRICT="mirror strip bindist"
 
-LICENSE="Microsoft-VSCode"
+LICENSE="
+	Microsoft-VSCode
+	0BSD
+    AFL-1.1
+    AFL-1.2
+    AFL-2
+    AFL-2.1
+    AFL-3
+    AGPL-1
+    AGPL-3
+    APSL-1
+    APSL-1.1
+    APSL-1.2
+    APSL-2
+    Apache-1
+    Apache-1.1
+    Apache-2
+    Artistic-1-Perl
+    Artistic-1-cl8
+    Artistic-1
+    Artistic-2
+    BCL
+    BSD-2-Clause-FreeBSD
+    BSD-2-Clause-NetBSD
+    BSD-2-Clause
+    BSD-3-Clause-Attribution
+    BSD-3-Clause-Clear
+    BSD-3-Clause-LBNL
+    BSD-3-Clause
+    BSD-4-Clause-UC
+    BSD-4-Clause
+    BSD-Protection
+    BSL-1
+    Beerware
+    CC-BY-1
+    CC-BY-2
+    CC-BY-2.5
+    CC-BY-3
+    CC-BY-4
+    CC-BY-NC-1
+    CC-BY-NC-2
+    CC-BY-NC-2.5
+    CC-BY-NC-3
+    CC-BY-NC-4
+    CC-BY-NC-ND-1
+    CC-BY-NC-ND-2
+    CC-BY-NC-ND-2.5
+    CC-BY-NC-ND-3
+    CC-BY-NC-ND-4
+    CC-BY-NC-SA-1
+    CC-BY-NC-SA-2
+    CC-BY-NC-SA-2.5
+    CC-BY-NC-SA-3
+    CC-BY-NC-SA-4
+    CC-BY-ND-1
+    CC-BY-ND-2
+    CC-BY-ND-2.5
+    CC-BY-ND-3
+    CC-BY-ND-4
+    CC-BY-SA-1
+    CC-BY-SA-2
+    CC-BY-SA-2.5
+    CC-BY-SA-3
+    CC-BY-SA-4
+    CC0-1
+    CDDL-1
+    CDDL-1.1
+    CPAL-1
+    CPL-1
+    Commons-Clause
+    EPL-1
+    EPL-2
+    EUPL-1
+    EUPL-1.1
+    FTL
+    Facebook-2-Clause
+    Facebook-3-Clause
+    Facebook-Examples
+    FreeImage
+    GPL-1
+    GPL-2-with-GCC-exception
+    GPL-2-with-autoconf-exception
+    GPL-2-with-bison-exception
+    GPL-2-with-classpath-exception
+    GPL-2-with-font-exception
+    GPL-2
+    GPL-3-with-GCC-exception
+    GPL-3-with-autoconf-exception
+    GPL-3
+    GUST-Font-License
+    IPL-1
+    ISC
+    ImageMagick
+    LGPL-2
+    LGPL-2.1
+    LGPL-3
+    LGPLLR
+    LPL-1
+    LPL-12
+    LPPL-1.3c
+    Libpng
+    Lil-1
+    Linux-OpenIB
+    MIT
+    MPL-1
+    MPL-1.1
+    MPL-2
+    MS-PL
+    NCSA
+    NPL-1
+    NPL-1.1
+    OFL-1.1
+    OSL-1
+    OSL-1.1
+    OSL-2
+    OSL-2.1
+    OSL-3
+    OpenSSL
+    OpenVision
+    PHP-3
+    PHP-31
+    PIL
+    PostgreSQL
+    Python-2-complete
+    Python-2
+    QPL-1
+    Ruby
+    SGI-B-1
+    SGI-B-1.1
+    SGI-B-2
+    SISSL-1.2
+    SISSL
+    Sleepycat
+    UPL-1
+    Unicode-DFS-2015
+    Unicode-DFS-2016
+    Unicode-TOU
+    Unlicense
+    W3C-19980720
+    W3C-20150513
+    W3C
+    WTFPL
+    X11
+    Xnet
+    ZPL-1.1
+    ZPL-2
+    ZPL-2.1
+    Zend-2
+    Zlib
+    eGenix
+    zlib-acknowledgement
```

Comment:
I'd honestly just go through the file with `licenseclassifier` and add those with **high** confidence. Then match those close to each other. It's a good start for us. This list seems unnaturally long. 

Add the chromium license though.

Reply:
How have you obtained that list, anyway? For example, where did you find GPL-2-with-bison-exception in that package?

Reply:
Noted @juippis, I made the mistake of using `license_serializer` instead of `identify_license`. I just wrote `license` in my bash and hit TAB and it got auto-completed to `license_serializer` and I though "fair enough, let's test it on `/opt/vscode/resources/app/ThirdPartyNotices.txt` @ulm that's how I obtained the list.

A quick try of `identify_license` missed a CC-BY-SA short license for a logo :thinking:

Reply:
I fear there will be no way around manually going through the licenses list provided by upstream. You can use an automated tool as a starting point, but entries like `GPL-1` or `GPL-2-with-bison-exception` indicate that the result must be double checked.

Reply:
Each license in `ThirdPartyNotices.txt` is well-separated with a `NOTICES AND INFORMATION BEGIN HERE` and `END OF [...] NOTICES AND INFORMATION`. There are 438 licenses, I will see what I can do :thinking:

Reply:
The work to write down all the licenses is really too much work... I am on the brink of giving up.

This license just doesn't have a name
```
%% aproba NOTICES AND INFORMATION BEGIN HERE
=========================================
Copyright (c) 2015, Rebecca Turner <me@re-becca.org>

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted, provided that the above
copyright notice and this permission notice appear in all copies.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
=========================================
END OF aproba NOTICES AND INFORMATION
```

Reply:
What to do with this Chromium license:
```
// Copyright 2014 The Chromium Authors. All rights reserved.
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
//
//    * Redistributions of source code must retain the above copyright
// notice, this list of conditions and the following disclaimer.
//    * Redistributions in binary form must reproduce the above
// copyright notice, this list of conditions and the following disclaimer
// in the documentation and/or other materials provided with the
// distribution.
//    * Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
The important part is this:
```
Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
```
I do not think it's in the regular BSD license.

Reply:
Another license I cannot classify but for sure it must be somewhere
```
%% entities NOTICES AND INFORMATION BEGIN HERE
=========================================
Copyright (c) Felix Böhm
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

THIS IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS,
EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
=========================================
END OF entities NOTICES AND INFORMATION
```

Reply:
How about this one ?
```
%% fadeevab/make.tmbundle NOTICES AND INFORMATION BEGIN HERE
=========================================
Copyright (c) textmate-make.tmbundle project authors

If not otherwise specified (see below), files in this repository fall under the following license:

Permission to copy, use, modify, sell and distribute this
software is granted. This software is provided "as is" without
express or implied warranty, and with no claim as to its
suitability for any purpose.

An exception is made for files in readable text which contain their own license information,
or files where an accompanying file exists (in the same directory) with a "-license" suffix added
to the base-name name of the original file, and an extension of txt, html, or similar. For example
"tidy" is accompanied by "tidy-license.txt".
=========================================
END OF fadeevab/make.tmbundle NOTICES AND INFORMATION
```

Reply:
This is BSD-1 right ?
```
%% split2 NOTICES AND INFORMATION BEGIN HERE
=========================================
Copyright (c) 2014-2018, Matteo Collina <hello@matteocollina.com>

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted, provided that the above
copyright notice and this permission notice appear in all copies.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
=========================================
END OF split2 NOTICES AND INFORMATION
```

Reply:
> The work to write down all the licenses is really too much work... I am on the brink of giving up.
> 
> This license just doesn't have a name
> 
> ```
> %% aproba NOTICES AND INFORMATION BEGIN HERE
> =========================================
> Copyright (c) 2015, Rebecca Turner <me@re-becca.org>
> 
> Permission to use, copy, modify, and/or distribute this software for any
> purpose with or without fee is hereby granted, provided that the above
> copyright notice and this permission notice appear in all copies.
> 
> THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
> WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
> MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
> ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
> WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
> ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
> OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
> =========================================
> END OF aproba NOTICES AND INFORMATION
> ```

ISC

Reply:
> What to do with this Chromium license:
> 
> ```
> // Copyright 2014 The Chromium Authors. All rights reserved.
> //
> // Redistribution and use in source and binary forms, with or without
> // modification, are permitted provided that the following conditions are
> // met:
> //
> //    * Redistributions of source code must retain the above copyright
> // notice, this list of conditions and the following disclaimer.
> //    * Redistributions in binary form must reproduce the above
> // copyright notice, this list of conditions and the following disclaimer
> // in the documentation and/or other materials provided with the
> // distribution.
> //    * Neither the name of Google Inc. nor the names of its
> // contributors may be used to endorse or promote products derived from
> // this software without specific prior written permission.
> //
> // THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
> // "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
> // LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
> // A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
> // OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
> // SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
> // LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
> // DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
> // THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
> // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
> // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> ```
> 
> The important part is this:
> 
> ```
> Neither the name of Google Inc. nor the names of its
> // contributors may be used to endorse or promote products derived from
> // this software without specific prior written permission.
> ```
> 
> I do not think it's in the regular BSD license.
> 
> Edit: Okay it's BSD-4

It's regular (3-clause) BSD.

Reply:
Alright thanks! I am nearly done going through every license. The big savior is that it's filled with MIT licenses. I am using a regex to isolate each license. After removing all the MIT occurences.

Reply:
How about this one ?
```
%% textmate/asp.vb.net.tmbundle NOTICES AND INFORMATION BEGIN HERE
=========================================
Copyright (c) textmate-asp.vb.net.tmbundle project authors

If not otherwise specified (see below), files in this folder fall under the following license:

Permission to copy, use, modify, sell and distribute this
software is granted. This software is provided "as is" without
express or implied warranty, and with no claim as to its
suitability for any purpose.

An exception is made for files in readable text which contain their own license information,
or files where an accompanying file exists (in the same directory) with a "-license" suffix added
to the base-name name of the original file, and an extension of txt, html, or similar. For example
"tidy" is accompanied by "tidy-license.txt".
=========================================
END OF textmate/asp.vb.net.tmbundle NOTICES AND INFORMATION
```

Thank you for your help!